doc/examples/combined_workflow_creation.py

```diff
@@ -0,0 +1,143 @@
+"""
+============================================================
+Creating a new combined workflow.
+============================================================
+
+A combined workflow is a series of dipy workflows organized together in a way
+that the output of a workflow serves as input for the next one.
+"""
+
+"""
+First create your combined workflow class. Your combine workflow class file
+is usually located in the ../dipy/workflows directory.
+"""
+
+from dipy.workflows.combined_workflow import CombinedWorkflow
+
+"""
+``CombinedWorkflow`` is the base class that will be extended to create our
+combined workflow.
+"""
+
+from dipy.workflows.denoise import  NLMeansFlow
+from dipy.workflows.segment import MedianOtsuFlow
+
+"""
+``MedianOtsuFlow`` and ``NLMeansFlow`` will be combined to create our
+processing section.
+"""
+
+
+class DenoiseAndSegment(CombinedWorkflow):
+
+    """
+    ``DenoiseAndSegment`` is the name of our combined workflow. Note that
+    it needs to extend CombinedWorkflow for everything to work properly.
+    """
+
+    def _get_sub_flows(self):
+        return [
+            NLMeansFlow,
+            MedianOtsuFlow
+        ]
+
+        """
+        It is mandatory to implement this method if you want to make all the sub
+        workflows parameters available in commandline.
+        """
+
+    def run(self, input_files, out_dir='', out_file='processed.nii.gz'):
+        """
+        Parameters
+        ----------
+        input_files : string
+            Path to the input files. This path may contain wildcards to
+            process multiple inputs at once.
+
+        out_dir : string, optional
+            Where the resulting file will be saved. (default '')
+
+        out_file : string, optional
+            Name of the result file to be saved. (default 'processed.nii.gz')
+        """
+
+        """
+        Just like a normal workflow, it is mandatory to have out_dir as a
+        parameter. It is also mandatory to put 'out_' in front of every
+        parameter that is going to be an output. Lastly, all out_ params needs
+        to be at the end of the params list.
+
+        The class docstring part is very important, you need to document
+        every parameter as they will be used with inspection to build the
+        command line argument parser.
+        """
+
+        io_it = self.get_io_iterator()
+
+        for in_file, out_file in io_it:
+            nl_flow = NLMeansFlow()
+            self.run_sub_flow(nl_flow, in_file, out_dir=out_dir)
+            denoised = nl_flow.last_generated_outputs[0][0]
```

Comment:
Last generated outputs is a list of all file paths generated by the last run of this workflow. Since the paths are generated when workflow.run is called (via io_iterator) this can only be retreived after workflow.run was called. This is how you link a workflow's output to the next workflow's input.

Since last_generated_outputs is an io_iterator.outputs its a list of list. Hence the [0][0]. Ill try to refactor that part so it feels more user friendly.